### PR TITLE
Refactor extract PasswordEdit to 'gui.base' namespace

### PR DESCRIPTION
**Pull request by gonzalo-bulnes**
# Description

This is the third step to be split out of #1369.
It moves the `PasswordEdit` definition to its own file/module. Eventually, more form inputs could be extracted there, but for now there is only this one.

This step is necessary to prevent a circular dependency when extracting the `LoginDialog` out of `widgets.py`.

# Test Plan

- [ ] Confirm that the passphrase input in the login dialog behavior and appearance haven't changed
- [ ] Confirm that the `PasswordEdit` definition is unchanged
- [ ] Confirm that the `PasswordEdit` test is unchaged and passes
- [ ] Confirm that CI is green

# Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes is required. Please check as applicable:

 - [ ] I have tested these changes in the appropriate Qubes environment
 - [ ] I do not have an appropriate Qubes OS workstation set up (the reviewer will need to test these changes)
 - [x] These changes should not need testing in Qubes

If these changes add or remove files other than client code, the AppArmor profile may need to be updated. Please check as applicable:

 - [ ] I have updated the [AppArmor profile](https://github.com/freedomofpress/securedrop-client/blob/HEAD/files/usr.bin.securedrop-client)
 - [x] No update to the AppArmor profile is required for these changes
 - [ ] I don't know and would appreciate guidance

If these changes modify the database schema, you should include a database migration. Please check as applicable:

 - [ ] I have written a migration and upgraded a test database based on `main` and confirmed that the migration applies cleanly
 - [ ] I have written a migration but have not upgraded a test database based on `main` and would like the reviewer to do so
 - [ ] I need help writing a database migration
 - [x] No database schema changes are needed

### CODE REVIEW

**Comment by gonzalo-bulnes** [comment]
@creviera Before merging this PR, let's talk about signatures. It seems that the author signature gets stripped when merging without a merge commit, resulting in an unsigned commit in `main`. We could put extra attention on this one and see if we can merge while preserving the signatures on the commits that introduce diffs. (The automated merge commit should be empty because we update/rebase out branches before merging.)

**Comment by rocodes** [comment]
@gonzalo-bulnes just double-checking cause of the title- you meant the `gui.inputs` namespace rather than `gui.base` namespace?

**Comment by gonzalo-bulnes** [comment]
@rocodes Great question! I meant 'gui.base', but that package was introduced in #1382 and I ended up deciding to assume we'd merge that one and rebase this PR before reviewing it. (That's where PRs made of one commit are get less practical if you ask me.) Ultimately `gui.base` is where `PasswordEdit` should be accessible from and imported from -but currently the diff makes it accessible and imported from `gui`- does that make sense?

Note I say `gui`, not `gui.inputs`: the _module_ is called `inputs`, but that module belongs to the `securedrop_client.gui` (`gui` for short, later `gui.base`) _package_ (that has a `__init__` file) and is what's components are imported from.

File: tests/gui/base/test_inputs.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+app = QApplication([])
+
+
+def test_PasswordEdit(mocker):
+    passwordline = PasswordEdit(None)
+    passwordline.togglepasswordAction.trigger()
+
+    assert passwordline.echoMode() == QLineEdit.Normal
+    passwordline.togglepasswordAction.trigger()
+    assert passwordline.echoMode() == QLineEdit.Password
```
Comment (conorsch) [inline tests/gui/base/test_inputs.py:14]: Test is ported to its new location, but not removed from its old. Fix and squash in?

**Review by conorsch — APPROVED** [review]
Thanks for the test tweak; lgtm. I'm approving, but not merging, because we have an unrelated CI failure right now. Will bounce CI later and try to get it to green.

**Review by conorsch — CHANGES_REQUESTED** [review]
CI is now failing a valid lint suggestion! @gonzalo-bulnes please take another look. (Normally I'd fix and approve, but given all the rebasing you're about to do, figured it's best that you always have the latest branch locally!)

**Comment by gonzalo-bulnes** [comment]
Yes! Oops, I missed the linting step indeed. Fixing that.

**Comment by gonzalo-bulnes** [comment]
(amend) Remove `QLineEdit` import from `test_widgets.py`. :white_check_mark:

**Review by rocodes — APPROVED** [review]
Looks good to me, `make test` passes locally, login dialog and show/hide password functionality are unchanged, visual review is straightforward :)

(Noting that `make check` fails for me locally, but the issue is unrelated to this PR and I see green on CI so I'm approving)

**Review by conorsch — APPROVED** [review]

**Pull request merged**